internal/pkg/cli/task_run.go

```diff
@@ -500,6 +580,12 @@ func (o *runTaskOpts) Ask() error {
 			return err
 		}
 	}
+	if len(o.secrets) > 0 {
+		o.categorizeSecrets()
```

Comment:
We should do this after `confirmSecretsAccess` (would be best to place in `Execute()` so that we don't need to define extra member variables `ssmParamSecrets` and `secretsManagerSecrets`). Otherwise it might happen that both (`App`, `Env`) and (`ssmParamSecrets`, `secretsManagerSecrets`) are set.

Reply:
I think this is done during `Ask` because we can reuse the categorized variable during `Ask` (when we prompts, we list out all ssm secrets, and then all secretsmanager secrets). 

> Otherwise it might happen that both (App, Env) and (ssmParamSecrets, secretsManagerSecrets) are set.

I wonder why is this an issue though 🤔 ? If `App` `Env` can co-exist with `Secrets`, why can't they with `SSMParam` and `SecretsManagerSecrets`?

Reply:
Because https://github.com/aws/copilot-cli/pull/3288/files#diff-ee1122ecbf021b404a3085486571ea2beb4dff929bbe256e99b852c18f99dd16R100 and https://github.com/aws/copilot-cli/pull/3288/files#diff-ee1122ecbf021b404a3085486571ea2beb4dff929bbe256e99b852c18f99dd16R125 they have the same policy name making me feel they are contradict with each other and only one set exists at a time

Reply:
But we now have else condition [here](https://github.com/aws/copilot-cli/pull/3288/files#diff-ee1122ecbf021b404a3085486571ea2beb4dff929bbe256e99b852c18f99dd16R119) so either app and env block will be executed or `SSMParam` and `SecretsManagerSecrets` block will be executed.

Reply:
Gotcha yeah i know it wouldn't be a real issue. It's just `o.categorizeSecrets()` looks very stateful to me.

Reply:
Alternatively, we could consider caching it and have `categorizeSecrets` to return a struct of SSM and secret manager. So when the cache var is not `nil` we just return the cache var. Otherwise, we try to categorize and return the var. Purpose for this is to make it not that stateful: whenever we need these vars we call the function.

Reply:
```
func (o *runTaskOpts) getCategorizedSecrets() {
	if o.ssmParamSecrets != nil || o.secretsManagerSecrets != nil {
                 return o.ssmParamSecrets, o.secretsManagerSecrets
          }
	for name, value := range o.secrets {
		if isSSM(value){
			o.ssmParamSecrets[name] = value
		}
		if isSecretsManager(VALUE) {
			o.secretsManagerSecrets[name] = value
		}
	}
        return o.ssmParamSecrets, o.secretsManagerSecrets 
}

func (o *runTaskOpts) confirmSecretsAccess() error {
   ssm, secrets := o.getCategorizedSecrets()
}

func (o *runTaskOpts) Execute() error {
   ssm, secrets := o.getCategorizedSecrets()
}
```